### ISSUE TITLE
ERROR](CurlHttpClient)Curl returned error code 28

### ISSUE THREAD

**Issue by muhufuk** [awslabs/amazon-kinesis-producer#268]
I have seen following logs on our applicaiton's logs. Our application runs on EC2 (m4xlarge)

[ERROR] [ ] [on-0003] [c.a.services.kinesis.producer.LogInputStreamReader]: [2019-07-22 14:30:24.972547] [0x00000052][0x00007f56a3fff700] [error] [AWS Log: ERROR](CurlHttpClient)Curl returned error code 28

I would like to know what is the reason of exception above.

KPL version is 0.12.7

**Comment by ramtg** [comment]
I see this error message in production application logs.

Please advise as to how to handle this error.

Thank you.

**Comment by MihirLimbachia** [comment]
Curl 28 indicates a request timed out.

I would recommend increasing MaxConnections parameter gradually in KPL and check if the errors persist.

https://github.com/awslabs/amazon-kinesis-producer/blob/e222115a91c51c75430e91b749591dd82ba53e2f/java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducerConfiguration.java#L1035

**Comment by lakshmikanthpadala** [comment]
@muhufuk @ramtg Did change in max connections helped?

**Comment by DanOrsborne** [comment]
Did this solve the issue?

**Comment by DanOrsborne** [comment]
> Configure RequestTimeout property of KPL to the higher value

No!  That's not a fix.  That just makes it wait longer before failing

**Comment by Mariani88** [comment]
Could you resolve this issue? I am receiving the same error
